opendatahub-io/kserve
# Code sync up to upstream commit for v0.11.1

## Testing

Based on instructions of the following PR: https://github.com/opendatahub-io/kserve/pull/93:

A simple sanity check should be OK.

Checkout this PR:

1. Clone the repo: `git clone https://github.com/opendatahub-io/kserve.git && cd kserve`
2. Fetch the PR: `git fetch origin pull/108/merge:pr-108`
3. Checkout the code: `git checkout pr-108`

On a clean OpenShift cluster, and given you have the code of this PR checked out:

1. Deploy ServiceMesh. Run: `test/scripts/openshift-ci/deploy.ossm.sh`
2. Deploy Serverless. Run: `test/scripts/openshift-ci/deploy.serverless.sh`
3. Deploy KServe:
```sh
oc new-project kserve
oc label namespace kserve testing.kserve.io/add-to-mesh=true
kustomize build config/overlays/odh | \
  sed "s|kserve/storage-initializer:latest|quay.io/opendatahub/kserve-storage-initializer:pr-108|" | \
  sed "s|kserve/agent:latest|quay.io/opendatahub/kserve-agent:pr-108|" | \
  sed "s|kserve/router:latest|quay.io/opendatahub/kserve-router:pr-108|" | \
  sed "s|kserve/kserve-controller:latest|quay.io/opendatahub/kserve-controller:pr-108|" | \
  oc apply -f -
```
4. Deploy a test model:
```sh
oc new-project kserve-test
oc label namespace kserve-test testing.kserve.io/add-to-mesh=true
sed 's/ClusterServingRuntime/ServingRuntime/' config/runtimes/kserve-mlserver.yaml | \
  sed "s|mlserver:replace|docker.io/seldonio/mlserver:1.3.2|" | \
  oc apply -f -
cat <<EOF | oc apply -f -
apiVersion: "serving.kserve.io/v1beta1"
kind: "InferenceService"
metadata:
  name: "sklearn-irisv2"
  annotations:
    serving.knative.openshift.io/enablePassthrough: "true"
    sidecar.istio.io/inject: "true"
    sidecar.istio.io/rewriteAppHTTPProbers: "true"
spec:
  predictor:
    model:
      modelFormat:
        name: sklearn
      runtime: kserve-mlserver
      storageUri: "gs://seldon-models/sklearn/mms/lr_model"
EOF
```
5. Check that the model replies correctly:
```sh
ENDPOINT=$(oc get ksvc sklearn-irisv2-predictor -o jsonpath='{.status.url}')
curl -kv -H "Content-Type: application/json" -d '{"inputs": [{"name": "input-0","shape": [2, 4],"datatype": "FP32","data": [[6.8, 2.8, 4.8, 1.4],[6.0, 3.4, 4.5, 1.6]]}]}' $ENDPOINT/v2/models/sklearn-irisv2/infer
```

If you get a reply, it should all be running OK.
